### ISSUE TITLE
Auto update from 0.3.0 and 0.3.1 does not work on Windows because of failed certificate check

### ISSUE THREAD

**Issue by QueensDeeJay** [tkainrad/keycombiner#26]
Error:
```Error: New version 0.3.1 is not signed by the application owner: publisherNames: Developer ID Application: Thomas Kainrad (YXNM6Q6X29), raw info: {
  "SignerCertificate": {
    "FriendlyName": "",
    "IssuerName": {
      "Name": "C=US, O=Apple Inc., OU=Apple Certification Authority, CN=Developer ID Certification Authority",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "NotAfter": "/Date(1782672082000)/",
    "NotBefore": "/Date(1624819283000)/",
    "PrivateKey": null,
    "PublicKey": {
      "Key": "System.Security.Cryptography.RSACryptoServiceProvider",
      "Oid": "System.Security.Cryptography.Oid",
      "EncodedKeyValue": "System.Security.Cryptography.AsnEncodedData",
      "EncodedParameters": "System.Security.Cryptography.AsnEncodedData"
    },
    "SerialNumber": "31F570930BFC0DF7",
    "SignatureAlgorithm": {
      "Value": "1.2.840.113549.1.1.11",
      "FriendlyName": "sha256RSA"
    },
    "Thumbprint": "938D4254A86CF2514CEFB9EEA03F1ACB695CE177",
    "Version": 3,
    "Issuer": "C=US, O=Apple Inc., OU=Apple Certification Authority, CN=Developer ID Certification Authority",
    "Subject": "C=US, O=Thomas Kainrad, OU=YXNM6Q6X29, CN=Developer ID Application: Thomas Kainrad (YXNM6Q6X29), OID.0.9.2342.19200300.100.1.1=YXNM6Q6X29"
  },
  "TimeStamperCertificate": {
    "Archived": false,
    "Extensions": [
      "System.Security.Cryptography.X509Certificates.X509KeyUsageExtension",
      "System.Security.Cryptography.X509Certificates.X509BasicConstraintsExtension",
      "System.Security.Cryptography.X509Certificates.X509EnhancedKeyUsageExtension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509SubjectKeyIdentifierExtension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509Extension"
    ],
    "FriendlyName": "",
    "IssuerName": {
      "Name": "CN=DigiCert SHA2 Assured ID Timestamping CA, OU=www.digicert.com, O=DigiCert Inc, C=US",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "NotAfter": "/Date(1925424000000)/",
    "NotBefore": "/Date(1609459200000)/",
    "HasPrivateKey": false,
    "PrivateKey": null,
    "PublicKey": {
      "Key": "System.Security.Cryptography.RSACryptoServiceProvider",
      "Oid": "System.Security.Cryptography.Oid",
      "EncodedKeyValue": "System.Security.Cryptography.AsnEncodedData",
      "EncodedParameters": "System.Security.Cryptography.AsnEncodedData"
    },
    "SerialNumber": "0D424AE0BE3A88FF604021CE1400F0DD",
    "SubjectName": {
      "Name": "CN=DigiCert Timestamp 2021, O=\"DigiCert, Inc.\", C=US",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "SignatureAlgorithm": {
      "Value": "1.2.840.113549.1.1.11",
      "FriendlyName": "sha256RSA"
    },
    "Thumbprint": "E1D782A8E191BEEF6BCA1691B5AAB494A6249BF3",
    "Version": 3,
    "Handle": 2954766176464,
    "Issuer": "CN=DigiCert SHA2 Assured ID Timestamping CA, OU=www.digicert.com, O=DigiCert Inc, C=US",
    "Subject": "CN=DigiCert Timestamp 2021, O=\"DigiCert, Inc.\", C=US"
  },
  "Status": 1,
  "StatusMessage": "A certificate chain could not be built to a trusted root authority"
}
    at newError (C:\Users\deeja\AppData\Local\Programs\keycombiner\resources\app.asar\node_modules\builder-util-runtime\src\index.ts:29:17)
    at Object.task (C:\Users\deeja\AppData\Local\Programs\keycombiner\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:41:17)
    at NsisUpdater.executeDownload (C:\Users\deeja\AppData\Local\Programs\keycombiner\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:617:7)
```
OS: Windows NT 10.0.19043

**Comment by tkainrad** [comment]
Thanks a lot for reporting this problem. I could verify and reproduce it. Apparently, I did not test the update procedure from 0.3.0 to 0.3.1 on Windows. I installed 0.3.1 manually on Windows and tested this way.

The problem is that starting with 0.3.0, KeyCombiner Desktop is signed and notarized by Apple so that macOS Gatekeeper does allow the installation without any warnings. Now, when auto-updating, KeyCombiner tries to validate this certificate. However,  this validation fails on Windows because Windows does not have the required certification authority's certificate installed to validate Apple certificates.

There is a bit more background in [this issue](https://github.com/electron-userland/electron-builder/issues/1930) and [this issue](https://github.com/electron-userland/electron-builder/issues/1930) of other applications that ran into this problem a while ago.

**Unfortunately, I cannot make this problem go away for the upgrade from 0.3.0 on Windows. **

**Please uninstall 0.3.0 and manually download and install 0.3.1. I will make sure that upgrades from 0.3.1 to future versions will again work also on Windows.**

**Comment by Positron010** [comment]
Hi,
Apparently this issue still persists as of now. I encountered this issue when trying to upgrade form 0.3.0 to 0.3.1 inside the app, but resolved it by installing manually. I somehow forgot to report. Now I encountered it again when upgrading from 0.3.1 to 0.3.2 inside the app as well. The latest version available for download in the website is 0.3.1, which I installed assuming it has the issue resolved, but I also encountered the same issue.

Thanks for your efforts

**Comment by tkainrad** [comment]
Hi,

Sorry for the inconvenience. It turned out that manually installing 0.3.1 would not be enough as this would still lead to problems with future versions.

Therefore, I just now released 0.3.2 which fixes the Windows auto-updating behavior. It can now (since 5 minutes) be downloaded from the website at https://keycombiner.com/desktop/  
Please manually download and install this version one last time. The next update to a future version will be automated once again.

Version 0.3.2 also brings another frequently requested feature: Autostart on system login! It is enabled by default and will launch KeyCombiner Desktop in minimized state when you log in to Windows. This way, the instant lookup will be ready to use immediately :)

**Comment by tkainrad** [comment]
Closing this now as it has been resolved with 0.3.2 which is now officially released. If you are on Windows and are currently using 0.3.1 or 0.3.2, please manually update to 0.3.2 by removing your old installation and installing the new version.

0.3.2 also adds autostart support, see #25
